core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3067,44 +3067,56 @@ public static void setLocalDictColumnsToWrapperSchema(List<ColumnSchema> columns
         }
         // if complex column is defined in local dictionary include column, then get the child
         // columns and set the string datatype child type as local dictionary column
-        if (column.getNumberOfChild() > 0 && null != localDictIncludeColumnsOfMainTable) {
-          listOfDictionaryIncludeColumns = localDictIncludeColumnsOfMainTable.split(",");
+        if (column.getNumberOfChild() > 0 && null != localDictIncludeColumns) {
+          listOfDictionaryIncludeColumns = localDictIncludeColumns.trim().split("\\s*,\\s*");
```

Comment:
If possible move trim and split outside of loop as it output will be same and it is happening in loop which will impact the performance.